### DISCUSSION TITLE
Can we start red eclipse from command line with arguments?

### DISCUSSION THREAD

**Discussion by liuty10** [redeclipse/base#1163] [Help and Support]
Can we start red eclipse from command line with arguments? So that we can enter game scene  directly other than clicking on the GUI. If so, could anyone give some examples?

**Comment by Calinou** [comment]
The `-x` command-line switch can be used to run commands on start (see the [Cube 2: Sauerbraten reference](http://sauerbraten.org/docs/config.html#command_line_options)), but it doesn't seem to work when the command contains spaces.

```bash
# Starts a random map in deathmatch mode.
# There must be no spaces after `-x`
path/to/redeclipse.sh -x"deathmatch"

# Should start deathmatch on map `dutility`, but actually starts
# a random map in deathmatch mode as only the first word is read
path/to/redeclipse.sh -x"deathmatch dutility"
```

**Comment by liuty10** [comment]
Thank you very much. Yeah, it is always random map, but I really need a definite map for benchmarking purpose. Is there any solutions? Should I modify the source code for this part?

**Comment by Calinou** [comment]
You could write an `autoexec.cfg` file to `$HOME/.redeclipse` with the following contents:

```text
// The `sleep` instruction is required, otherwise the game will segfault
// when starting the map
sleep 500 [deathmatch dutility]
```

**Comment by liuty10** [comment]
This work!
I have noticed that the game is always in SpecTV mode before we right click the mouse button. Is there any way to enter game mode directly? My real question is: are the frames in SpecTV mode from a video or rendered in real time?

**Comment by Calinou** [comment]
> My real question is: are the frames in SpecTV mode from a video or rendered in real time?

They are rendered in real-time. However, the bot movement isn't deterministic, and I don't think the SpecTV camera movement is either.

If you need deterministic playback, I advise fetching a demo after playing a game (or on a multiplayer server) using `/listdemos` and `/getdemo N` (where `N` is a demo's number). Use `/demo <filename>` to play the demo (press <kbd>Tab</kbd> to complete file names).

**Comment by liuty10** [comment]
Thank you very much. These informations are really helpful. I will try that.
